src/main/kotlin/org/opensearch/reportsscheduler/model/ReportDefinition.kt

```diff
@@ -234,85 +234,85 @@ internal data class ReportDefinition(
+                    when (fieldName) {
+                        DURATION_TAG -> durationSeconds = Duration.parse(parser.text())
+                        FILE_FORMAT_TAG -> fileFormat = FileFormat.valueOf(parser.text())
+                        LIMIT_TAG -> limit = parser.intValue()
+                        HEADER_TAG -> header = parser.textOrNull()
+                        FOOTER_TAG -> footer = parser.textOrNull()
+                        TIME_FROM_TAG -> timeFrom = parser.textOrNull()
+                        TIME_TO_TAG -> timeTo = parser.textOrNull()
+                        else -> {
+                            parser.skipChildren()
+                            log.info("$LOG_PREFIX:Format Skipping Unknown field $fieldName")
+                        }
                     }
                 }
+                durationSeconds ?: throw IllegalArgumentException("$DURATION_TAG field absent")
+                fileFormat ?: throw IllegalArgumentException("$FILE_FORMAT_TAG field absent")
+                return Format(
+                    durationSeconds,
+                    fileFormat,
+                    limit,
+                    header,
+                    footer,
+                    timeFrom,
+                    timeTo
+                )
             }
-            durationSeconds ?: throw IllegalArgumentException("$DURATION_TAG field absent")
-            fileFormat ?: throw IllegalArgumentException("$FILE_FORMAT_TAG field absent")
-            return Format(
-                durationSeconds,
-                fileFormat,
-                limit,
-                header,
-                footer,
-                timeFrom,
-                timeTo
-            )
         }
-    }
 
-    /**
-     * {@inheritDoc}
-     */
-    override fun toXContent(builder: XContentBuilder?, params: ToXContent.Params?): XContentBuilder {
-        builder!!
-        builder.startObject()
-            .field(DURATION_TAG, duration.toString())
-            .field(FILE_FORMAT_TAG, fileFormat.name)
-        if (limit != null) builder.field(LIMIT_TAG, limit)
-        if (header != null) builder.field(HEADER_TAG, header)
-        if (footer != null) builder.field(FOOTER_TAG, footer)
-        builder.field(TIME_FROM_TAG, timeFrom)
-        builder.field(TIME_TO_TAG, timeTo)
-        builder.endObject()
-        return builder
+        /**
+         * {@inheritDoc}
+         */
+        override fun toXContent(builder: XContentBuilder?, params: ToXContent.Params?): XContentBuilder {
+            builder!!
+            builder.startObject()
+                .field(DURATION_TAG, duration.toString())
+                .field(FILE_FORMAT_TAG, fileFormat.name)
+            if (limit != null) builder.field(LIMIT_TAG, limit)
+            if (header != null) builder.field(HEADER_TAG, header)
+            if (footer != null) builder.field(FOOTER_TAG, footer)
+            builder.field(TIME_FROM_TAG, timeFrom)
+            builder.field(TIME_TO_TAG, timeTo)
```

Comment:
if we change it to
```suggestion
            if (timeFrom != null) builder.field(TIME_FROM_TAG, timeFrom)
            if (timeTo != null) builder.field(TIME_TO_TAG, timeTo)
```
we won't need to add https://github.com/opensearch-project/reporting/blob/70b1334c1661e901e2979f341b30d2cd3826cb1a/src/test/kotlin/org/opensearch/integTest/rest/OnDemandReportGenerationIT.kt#L40

although it would be good to cover parsing timeFrom and timeTo in tests

FYI @sumukhswamy

Reply:
@sumukhswamy can you please pick up the above suggestion when we add more tests for `timeFrom` and `timeTo`